lib/evaluate_board.rb

```diff
@@ -13,30 +15,36 @@ def initialize(board_gateway)
     ]
   end
 
-  def execute()
+  def execute
     return 'Player one wins' if win_for_player?('X')
     return 'Player two wins' if win_for_player?('O')
-    board = @board_gateway.fetch_board
-    return 'Game over' if board.compact.empty?
+    return 'Game over' if game_over?
 
     ''
   end
 
   private
 
+  def game_over?
+    board = @board_gateway.fetch_board
+    return false if board.nil?
+
+    board.compact.length == 9
+  end
+
   def win_for_player?(player)
     players_marks = []
```

Comment:
Maybe move `player_marks = []` above line 40 so it's together with where it's used?

Reply:
Yes, thank you.